core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/LimsUtils.java

```diff
@@ -459,22 +465,37 @@ public boolean test(T t) {
     return Arrays.stream(sets).flatMap(Collection::stream).collect(Collectors.toSet());
   }
 
-  public static String makeVolumeAndConcentrationLabel(Double volume, Double concentration, VolumeUnit volumeUnits,
+  public static String makeVolumeAndConcentrationLabel(BigDecimal volume, BigDecimal concentration, VolumeUnit volumeUnits,
       ConcentrationUnit concentrationUnits) {
     String strVolumeUnits = (volumeUnits == null ? "" : volumeUnits.getUnits());
     String strConcentrationUnits = (concentrationUnits == null ? "" : concentrationUnits.getUnits());
-    if (volume != null && volume != 0 && concentration != null && concentration != 0) {
-      return String.format("%.0f%s@%.0f%s", volume, strVolumeUnits, concentration, strConcentrationUnits);
+    if (volume != null && volume.compareTo(BigDecimal.ZERO) != 0 && concentration != null
+        && concentration.compareTo(BigDecimal.ZERO) != 0) {
+      return String.format(makeFormatString(volume) + "@" + makeFormatString(concentration), volume, strVolumeUnits, concentration,
+          strConcentrationUnits);
     }
-    if (volume != null && volume != 0) {
-      return String.format("%.0f%s", volume, strVolumeUnits);
+    if (volume != null && volume.compareTo(BigDecimal.ZERO) != 0) {
+      return String.format(makeFormatString(volume), volume, strVolumeUnits);
     }
-    if (concentration != null && concentration != 0) {
-      return String.format("%.0f%s", concentration, strConcentrationUnits);
+    if (concentration != null && concentration.compareTo(BigDecimal.ZERO) != 0) {
+      return String.format(makeFormatString(concentration), concentration, strConcentrationUnits);
```

Comment:
I think you can just use `toNiceString` here if you want. The lab didn't have very specific formatting requirements for the labels.

Reply:
GLT-3013 requests one decimal place if < 10 and none otherwise